packages/lucide-preact/src/createLucideIcon.ts

```diff
@@ -30,7 +31,7 @@ const createLucideIcon = (iconName: string, iconNode: IconNode): FunctionCompone
         width:  String(size),
         height: size,
         stroke: color,
-        ['stroke-width' as 'strokeWidth']: strokeWidth,
+        ['stroke-width' as 'strokeWidth']: scaleStrokeWidth ? strokeWidth : Number(strokeWidth) / (Number(size) / 24),
```

Comment:
Hmm, We should not round the numbers, because the `strokeWidth` should have a float.
According to the float precision we could use toFixed() to convert it to maximum decimals.
I can take a look to your alternative calculation.

Reply:
To fixed will always show decimals, e.g. `4.000`.